### ISSUE TITLE
[Bug]: Change<T> does not work correctly with null items

### ISSUE THREAD

**Issue by PhilipCavanagh** [reactivemarbles/DynamicData#692]
### Describe the bug 🐞

- When an item in a `SourceList<T>` is replaced, the creation of the `Change<T>` requires that the parameter `Optional<T> previous` has a value.
- If `T` is nullable, the previous value may be `null`.
- It is not possible to create an `Optional<T>` with a value of `null`.

Either `Change<T>` should mandate that `T` be `notnull`, or `Optional<T>` should not assume that `null` means "no value".

### Step to reproduce

```
var source = new SourceList<object?>();

using (source.Connect().Subscribe())
{
    source.Add(null);
    source.ReplaceAt(0, new object()); // This line throws an `ArgumentException` from `Change<T>` constructor
}
```

### Reproduction repository

https://github.com/PhilipCavanagh/DynamicData/blob/main/src/DynamicData.Tests/List/SourceListBug.cs

### Expected behavior

The single `null` item in the `SourceList` should be replaced with the new object.

### DynamicData Version

7.13.1

**Comment by RolandPheasant** [comment]
Not allowing nulls is by design.

**Comment by PhilipCavanagh** [comment]
In that case, could all the type parameters have `notnull` constraints added?

**Comment by glennawatson** [comment]
`notnull` makes sense to add. Makes sense it doesn't have it due to it being a newer C# feature.

**Comment by RolandPheasant** [comment]
Fixed by #706
